### ISSUE TITLE
[POST-INSTALL] Missing explanation at "Run the Modified GRUB Shell"

### ISSUE THREAD

**Issue by rowild** [dortania/bugtracker#168]
* Guide(s) in question
OpenCore Post-Install: Fix CFG Lock: Turning off CFG lock manually

* Link to page(s) with the issue
https://dortania.github.io/OpenCore-Post-Install/misc/msr-lock.html#turning-off-cfg-lock-manually

* Clearly explain the issue(s) with the page(s) (if there is a visual issue, a screenshot is required)
Here the docu suddenly says:
"Run the Modified GRUB Shell"
There is no explanation anywhere how to do that.

* Include possible fixes
I did find a walkthrough on how to do that on another page that deals with hackintosh and would have asked if linking to that very post would be an option. But the source is from a hackintosher that you do not want to work with (_starts with a "t", "on" which there sits a "y"..._).
Copy & pasting would be unfair to the original poster. So what can be done here? How can this "leak" of documentation be filled?

**Comment by dreamwhite** [comment]
You're right, probably most users may be lead to confusion while reading the guide.
I suggest to write a statement like:

> Add [modGRUBShell.efi](https://github.com/datasone/grub-mod-setup_var/releases/latest) onto OC/Tools directory and include it accordingly in config.plist

**Comment by rowild** [comment]
What is missing, still, is that there has to be a restart into OpenCore, right? So far, the documentation explains steps that are taken on a running OS. But modGRUBShell.efi has to be executed somewhere else...
I actually still do not know how, because the guide I followed explains how to prepare a USB stick that has to be loaded on startup (meaning: Not loading OC or anything else, but just that very script).

**Comment by dreamwhite** [comment]
> What is missing, still, is that there has to be a restart into OpenCore, right? So far, the documentation explains steps that are taken on a running OS. But modGRUBShell.efi has to be executed somewhere else...
> I actually still do not know how, because the guide I followed explains how to prepare a USB stick that has to be loaded on startup (meaning: Not loading OC or anything else, but just that very script).

Mh, I got your point.
Answering your question: yes, after you found your CFG Lock offset (same applies for DVMT-Preallocated and DVMT-Total Gfx offsets), you'll need to reboot onto OpenCore after adding modGRUBShell.efi tool.
I don't wanna spam here a repo I created, but it contains a fully debug OpenCore EFI containing all the necessary tools that you may need to create your OpenCore EFI. 

TL;DR: after you found your CFG Lock offset, add modGRUBShell.efi onto your EFI (both in OC/Tools and OC/config.plist), reboot in OpenCore and launch modGRUBShell.efi

**Comment by rowild** [comment]
Since I am still not successful in starting up OC, I would still like to know - just from reading the docu - _how_ to start the modGRUBShell.efi? (I know: to a pro this seems too much of a detail, but it actually isn't. Those step-by-step explanations can be found in the really excellent install guide.)

I assume there will be selection list somewhere, where I can choose from. And this will probably open a shell, where I enter the "setup_var 0x4E 0x00"? (Maybe with a screenshot?)

IMO that would make the docu better understandable in the sense, that I as a reader can follow it even though I do not have OC running or a computer nearby. (And again: I say this because the installation guide is actually written that way [and it is really written very well!]: I can read it on the tramway to familiarise myself with it; and as soon as I am at the computer I can start working right away...)

**Comment by dreamwhite** [comment]
> Since I am still not successful in starting up OC, I would still like to know - just from reading the docu - _how_ to start the modGRUBShell.efi? (I know: to a pro this seems too much of a detail, but it actually isn't. Those step-by-step explanations can be found in the really excellent install guide.)
> 
> I assume there will be selection list somewhere, where I can choose from. And this will probably open a shell, where I enter the "setup_var 0x4E 0x00"? (Maybe with a screenshot?)

Well yes, you'll reach a selection list (boot-picker) where you can choose which entry boot (in our case modGRUBShell.efi). This will open a shell where you can enter the command.

I'll try to make a PR if is well accepted by dortania's team where it'll be explained step-by-step how to reach modGRUBShell.efi.

> Since I am still not successful in starting up OC
What you mean by that? Do you mean that you can reach OpenCore boot-picker but CAN'T boot macOS or you can't even boot OpenCore using a thumb drive formatted in FAT32?

Below a picture which depicts what does OpenCore builtin boot-picker looks like:

![image](https://user-images.githubusercontent.com/17933708/110221859-d2cdd680-7ece-11eb-89f8-395378b2722a.png)

Anyways, until me or whoever, will make the PR, here you are the steps you need to follow in order to open modGRUBShell.efi. I assume that at least you can reach the picker (boot-selection menu) using your OpenCore EFI.

1. Download [modGRUBShell.efi](https://github.com/datasone/grub-mod-setup_var/releases/latest) and place it into your OpenCore EFI Tools folder (EFI/OC/Tools).
2. Open your config.plist with ProperTree and do a snapshot using CorpNewt's ProperTree's OC-Snapshot feature (press Command-R and select your EFI folder). This will add the tool in the config (Misc/Tools)
3. Boot OpenCore and press the number/letter on your keyboard corresponding to modGRUBShell.efi and write the command for unlocking CFG Lock

If you still need help don't hesitate to write here

**Comment by rowild** [comment]
Thank you, @dreamwhite ! I think this would help the docu very much!

ATM I do not need help with this topic, since I actually solved it. However, I still have troubles getting my setup running. But I try to get feedback for those things on Discord, since github is not the place for that, as you probably will agree. Maybe I see you there? Could really use your expertise :-)

**Comment by dreamwhite** [comment]
> Thank you, @dreamwhite ! I think this would help the docu very much!
> 
> ATM I do not need help with this topic, since I actually solved it. However, I still have troubles getting my setup running. But I try to get feedback for those things on Discord, since github is not the place for that, as you probably will agree. Maybe I see you there? Could really use your expertise :-)

Sure, my nickname is`dreamwhite#0362`
